x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/trusted_apps_page.test.tsx

```diff
@@ -296,4 +370,136 @@ describe('When on the Trusted Apps Page', () => {
       });
     });
   });
+
+  describe('and there are no trusted apps', () => {
+    const releaseExistsResponse: jest.MockedFunction<
+      () => Promise<GetTrustedListAppsResponse>
+    > = jest.fn(async () => {
+      return {
+        data: [],
+        total: 0,
+        page: 1,
+        per_page: 1,
+      };
+    });
+    const releaseListResponse: jest.MockedFunction<
+      () => Promise<GetTrustedListAppsResponse>
+    > = jest.fn(async () => {
+      return {
+        data: [],
+        total: 0,
+        page: 1,
+        per_page: 20,
+      };
+    });
+
+    beforeEach(() => {
+      // @ts-ignore
+      coreStart.http.get.mockImplementation(async (path, options) => {
+        if (path === TRUSTED_APPS_LIST_API) {
+          const { page, per_page: perPage } = options.query as { page: number; per_page: number };
+
+          if (page === 1 && perPage === 1) {
```

Comment:
Again, not a big fan of putting too much fake logic in these mocks. Probably better patterns to accomplish this

Reply:
Not sure there is a better way here. We call the same API twice - one with "normal" arguments for populating the list, and another one to check if entries exist. This check here is to ensure we can identify both of them and are able to control the response for each in the test case.

Reply:
I should mention: there is another way to mock the responses without having the logic, and that's to use `.mockResolvedValueOnce()` for each sequential API call that is made. This felt more brittle to me because if any `http.get()` call is inserted at some point in the future, the test would break, so I opted for mocking the entire implementation in order to have the ability to inspect the call's arguments